harness/determined/pytorch/deepspeed/dsat/_run_dsat.py

```diff
@@ -1,41 +1,52 @@
-import argparse
 import logging
 import os
+import pathlib
+import pickle
 
 import determined as det
 from determined import searcher
-from determined.pytorch.deepspeed import get_ds_config_from_hparams, overwrite_deepspeed_config
 from determined.pytorch.deepspeed.dsat import _defaults, _utils
+from determined.util import merge_dicts
 
 
-def get_parsed_args() -> argparse.Namespace:
-    parser = argparse.ArgumentParser()
-    parser.add_argument("-c", "--config_path", type=str)
-    parser.add_argument("-md", "--model_dir", type=str)
-    parser.add_argument("-t", "--tuner-type", type=str, default="random")
-    args = parser.parse_args()
-    # Strip and only use the base names.
+def main(core_context: det.core.Context) -> None:
+    with pathlib.Path(_defaults.ARGS_PKL_PATH).open("rb") as f:
+        args = pickle.load(f)
+    # On-cluster, the relative paths to the below files just come from the base names.
     args.config_path = os.path.basename(args.config_path)
     args.model_dir = os.path.basename(args.model_dir)
-    return args
-
-
-def main(core_context: det.core.Context) -> None:
-    args = get_parsed_args()
-
-    submitted_config_dict = _utils.get_dict_from_yaml_or_json_path(args.config_path)
-
-    assert (
-        args.tuner_type in _defaults.ALL_SEARCH_METHOD_CLASSES
-    ), f"tuner-type must be one of {list(_defaults.ALL_SEARCH_METHOD_CLASSES)}, not {args.tuner_type}"
+    args.include = [os.path.basename(p) for p in args.include] if args.include is not None else []
 
-    search_method = _defaults.ALL_SEARCH_METHOD_CLASSES[args.tuner_type](
-        submitted_config_dict=submitted_config_dict,
-        model_dir=args.model_dir,
-    )
+    search_method = _utils.get_search_method_from_args(args)
     search_runner = searcher.RemoteSearchRunner(search_method, context=core_context)
 
-    search_runner.run(submitted_config_dict, model_dir=args.model_dir)
+    exp_config = _utils.get_dict_from_yaml_or_json_path(args.config_path)
+    # add the search runner's experiment id to the description of the corresonding Trial
+    exp_id = det.get_cluster_info()._trial_info.experiment_id
```

Comment:
I like the idea to make sure things are made more clear about which orchestrator connects to which experiment. I'm just wondering if it's kosher that we are accessing private methods in a chain of getattrs, perhaps there should be a public function for getting this. This is mostly a nit though

Reply:
Yeah, I would like the same, but couldn't remember a non-private way of getting the `exp_id`